docs/user-guide/multiple_sources.md

```diff
@@ -71,3 +71,107 @@ at that URL. If the `path` field is not set, Argo CD will use the repository sol
 !!! note
     Sources with the `ref` field set must not also specify the `chart` field. Argo CD does not currently support using  
     another Helm chart as a source for value files.
+
+## Copying files across sources
+
+In some cases the builtin helm, kustomize or directory processing cannot be used due to the utilization of Config Management Plugins (CMPs).
+In these cases the `helm.valueFiles` referencing of different sources cannot be used, therefore files has to be made accessible using different
+means. Currently this is being done by a `from:` section in the source.
+
+The `from:` section is a list with the following elements:
+
+```yaml
+from:
+  - sourcePath: $reporef/from/here
+    destinationPath: to/here
+    failOnOutOfBoundsSymlink: true
+```
+
+The meaning of the fields:
+
+1. `ref`: (required) This is the referenced repository, the referenced name has to be prefix with a dollar(`$`) sign. This repository will be the source of the copy operation.
+1. `sourcePath`: (required) The path relative to the referenced source's repository's root directory. The file or directory specified by this path will be copied.
+1. `destinationPath`: (required) The path relative to the current application's repository's directory, this will be the destination of the copy
+1. `failOnOutOfBoundsSymlink`: (optional) Marks the handling of out-of-bound symlinks. If `failOnOutOfBoundsSymlink` is `true` then processing will error out if any out-of-bound symlinks are encountered. If it's `false`, then these will be ignored, and a warning logged about them. Default is `false`.
```

Comment:
Im just a random who got interested in what could be in the next release, my first thought on reading this section of docs is 'what is out of bounds?',  outside of the $reporef ... or outside of the sourcePath.  How 'deep' is the symlink deref?  ie 'in bounds' -> 'in bounds' -> '../../../../../../etc/passwd'

Reply:
This functionality had been reused, out-of-bounds checking code was already part of argo. As this feature was not intotroduced by this patch, I don't think it should be explained here.

Otherwise the bound is the checkout's rootdir.